docs/specs/clients/sign/client-api.md

```diff
@@ -74,6 +74,50 @@ abstract class Client {
     reason: Reason;
   }): Promise<void>;
 
+  // ---------- Wallet Authentication Methods ----------------------------------------------- //
+
+  // for proposer to request wallet authentication
+  public abstract sessionAuthenticate(params: {
+    chains: string[];
+    domain: string;
+    nonce: string;
+    aud: string;
+    type?: CacaoHeader["t"];
+    nbf?: string;
+    exp?: string;
+    statement?: string;
+    requestId?: string;
+    resources?: string[];
+    expiry?: number;
+    pairingTopic?: string;
+    methods?: string[];
+  }): Promise<{
+      uri: string,
+      response: Promise<{
+        auths?: Cacao[];
```

Comment:
Why optional?

Reply:
Dynamic switcher e.g. approving normal sign session doesn't produce `auths`. Having it option or as empty `[]` array imo is the same